docs/http_api.rst

```diff
@@ -0,0 +1,1097 @@
+
+.. http:put:: /organizations/(int:organization_id)
+
+    Update an existing organization.
+
+    :<json str name: The name of the organization.
+    :>json int id: The id of the created organization.
+    :>json list teams: The list of teams that the organization has.
+    :>json list users: The list of users that are part of the organization.
+    :status 200: When the organization is successfully generated.
+    :status 400: When there is invalid information to update the organization.
+
+    **Example request**:
+
+    .. sourcecode:: http
+
+       PUT /organizations/4 HTTP/1.1
+       Content-Type: application/json
+
+       {"name": "Brotherhood Without Banners"}
+
+    **Example response**:
+
+    .. sourcecode:: http
+
+       HTTP/1.1 200 OK
+       Content-Type: application/json
+
+       {"name":"Brotherhood Without Banners","id":4,"teams":[],"url":"https://example.org/organizations/4","users":[]}
+
+.. http:delete:: /organizations/(int:organization_id)
+
+    Remove an existing organization.
+
+    :status 204: Resource successfully deleted
+
+   **Example request**:
+
+   .. sourcecode:: http
+
+      DELETE /organizations/4 HTTP/1.1
+
+   **Example response**:
+
+   .. sourcecode:: http
+
+      HTTP/1.1 204 No Content
+
+.. http:put:: /organizations/(int:organization_id)/user
+
+    Add a user to an existing organization.
+
+    :<json int user_id: The ID of the user to add.
+
+    :status 204: User was successfully added.
+
+    **Example request**:
+
+    .. sourcecode:: http
+
+        PUT /organizations/4/users/ HTTP/1.1
+        Content-Type: application/json
+
+        {"user_id": 2}
+
+    **Example response**:
+
+    .. sourcecode:: http
+
+        HTTP/1.1 204 No Content
+
+.. http:delete:: /organizations/(int:organization_id)/users/(int:user_id)
+
+    Remove a user from an organization.
+
+    :status 204: User was successfully removed from an organization
+
+    **Example request**:
+
+    .. sourcecode:: http
+
+        DELETE /organizations/4/users/2 HTTP/1.1
+
+    **Example response**:
+
+    .. sourcecode:: http
+
+        HTTP/1.1 204 No Content
+
+Teams
+^^^^^
+
+.. http:get:: /teams/
+
+    Get a list of all the teams you have read access to.
+
+    **Example request**:
+
+    .. sourcecode:: http
+
+        GET /teams/ HTTP/1.1
+
+    **Example response**:
+
+    .. sourcecode:: http
+
+        HTTP/1.1 200 OK
+
+        [
+            {
+                "id": 4,
+                "name": "admins",
+                "permissions": [],
+                "url": "https://example.org/teams/4",
+                "organization": 7
+            }
+        ]
+
+.. http:get:: /teams/
+
+    Allows filtering of teams to retreive a subset.
+
+    :query string permission_contains:
+        All the permission fields on the resulting teams must contain this string.
+    :query string object_id:
+        All the object_id fields on the resulting teams must equal this string.
```

Comment:
Do we need this query param if `permission_contains` allows us to filter on a permission's object id? Or is `permission_contains` for the permission `type` field only?